xgcm/grid.py

```diff
@@ -1268,19 +1268,46 @@ def set_metrics(self, key, value):
                 f"Metric axes {axes_not_found!r} not compatible with grid axes {tuple(self.axes)!r}"
             )
 
-        # initialize empty list
-        self._metrics[metric_axes] = []
-        for metric_varname in _maybe_promote_str_to_list(value):
-            if metric_varname not in self._ds:
-                raise KeyError(
-                    f"Metric variable {metric_varname} not found in dataset."
-                )
-            # resetting coords avoids potential broadcasting / alignment issues
-            metric_var = self._ds[metric_varname].reset_coords(drop=True)
+        metric_value = _maybe_promote_str_to_list(value)
+        existing_metric_axes = set(k for k in self._metrics.keys())
```

Comment:
```suggestion
        existing_metric_axes = set(self._metrics.keys())
```

should also work? `dict.keys()` and `dict.values()` are "iterables" so you can cast them to `tuple` `set` and `list` directly.